OnlineStore.BLL/Services/Implementations/CategoryService.cs

```diff
@@ -0,0 +1,69 @@
+﻿using AutoMapper;
+using FluentValidation;
+using OnlineStore.BLL.DTOs;
+using OnlineStore.BLL.Services.Interfaces;
+using OnlineStore.DAL.Entities;
+using OnlineStore.DAL.Repositories.Interfaces;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace OnlineStore.BLL.Services
+{
+    public class CategoryService : ICategoryService
+    {
+        private readonly IRepository<Category> _repository;
+        private readonly ICategoryRepository _categoryRepository;
+        private readonly IValidator<CategoryDto> _validator;
+        private readonly IMapper _mapper;
+
+        public CategoryService(IRepository<Category> repository, ICategoryRepository categoryRepository, IValidator<CategoryDto> validator, IMapper mapper)
+        {
+            _repository = repository;
+            _categoryRepository = categoryRepository;
+            _validator = validator;
+            _mapper = mapper;
+        }
+
+        public async Task<IEnumerable<CategoryDto>> GetAllCategoriesAsync()
+        {
+            var categories = await _repository.GetAllAsync();
+            return _mapper.Map<IEnumerable<CategoryDto>>(categories);
+        }
+
+        public async Task<CategoryDto> GetCategoryByIdAsync(int id)
+        {
+            var category = await _repository.GetByIdAsync(id);
```

Comment:
you need to write 
if(category is null)
{NotFoundException}
Review everywhere!!!